packages/ui/src/components/molecules/PopUp/NewPopUp.tsx

```diff
@@ -0,0 +1,161 @@
+import {Card, Text, Icon} from '../../atoms';
+import Button, {ButtonProps} from '../Button/Button';
+
+interface ButtonConfig extends ButtonProps {
+  hide?: boolean;
+}
+interface CancelButtonConfig extends ButtonConfig {
+  showInHeader?: boolean;
+}
+
+interface PopUpProps {
+  style?: any;
+  visible: boolean;
+  title?: string;
+  children: any;
+  cancelButtonConfig?: CancelButtonConfig;
+  confirmButtonConfig?: ButtonConfig;
+  translator?: (key: string) => string;
+}
+
+const PopUp = ({
+  style,
+  visible,
+  title,
+  children,
+  cancelButtonConfig,
+  confirmButtonConfig,
+  translator,
+}: PopUpProps) => {
+  const Colors = useThemeColor();
+
+  const _cancelButtonConfig = useMemo(() => {
+    if (!cancelButtonConfig) {
+      return null;
+    }
+
+    return {
+      ...cancelButtonConfig,
+      title: cancelButtonConfig.title ?? translator('Base_Cancel'),
+      color: cancelButtonConfig.color ?? Colors.errorColor,
+      iconName: cancelButtonConfig.iconName ?? 'times',
+      hide: cancelButtonConfig.hide ?? false,
+      showInHeader: cancelButtonConfig.showInHeader ?? false,
+      width: cancelButtonConfig.width ?? 115,
+    };
+  }, [cancelButtonConfig, translator, Colors]);
+
+  const _confirmButtonConfig = useMemo(() => {
+    if (!confirmButtonConfig) {
+      return null;
+    }
+
+    return {
+      ...confirmButtonConfig,
+      title: confirmButtonConfig.title ?? translator('Base_Ok'),
+      color: confirmButtonConfig.color ?? Colors.primaryColor,
+      iconName: confirmButtonConfig.iconName ?? 'check',
+      hide: confirmButtonConfig.hide ?? false,
+      width: confirmButtonConfig.width ?? 115,
+    };
+  }, [confirmButtonConfig, translator, Colors]);
+
+  return (
+    <Modal
+      visible={visible}
+      transparent
+      animationType="fade"
+      onRequestClose={_cancelButtonConfig?.onPress}>
+      <View style={styles.modalBackground}>
+        <Card style={[styles.container, style]}>
+          <View style={styles.headerContainer}>
+            {!!title && (
+              <Text writingType="title" fontSize={20}>
+                {translator ? translator(title) : title}
+              </Text>
+            )}
+            {!_cancelButtonConfig?.hide &&
+              _cancelButtonConfig?.showInHeader && (
+                <Icon
+                  name="times"
+                  size={20}
+                  style={styles.headerCancelButton}
+                  touchable
+                  onPress={_cancelButtonConfig?.onPress}
+                />
+              )}
+          </View>
+          {children}
+          <View style={styles.buttonsContainer}>
+            {!_cancelButtonConfig?.hide &&
+              !_cancelButtonConfig?.showInHeader && (
+                <Button {..._cancelButtonConfig} style={styles.cancelButton} />
+              )}
+            {!_confirmButtonConfig?.hide && (
+              <Button {..._confirmButtonConfig} />
+            )}
+          </View>
```

Comment:
If both buttons are hidden, this view should not be rendered as it will leave unwanted space